app/api/odm/ModelWithPermissions.ts

```diff
@@ -115,14 +121,15 @@ const controlPermissionsData = <T>(data: T & { published?: boolean }, user?: Use
 };
 
 export class ModelWithPermissions<T> extends OdmModel<T> {
-  async save(data: DataType<T & { permissions?: PermissionSchema[] }>) {
+  async save(data: WithPermissionsDataType<T>) {
     const user = permissionsContext.getUserInContext();
+    const query = { _id: data._id } as PermissionsUwaziFilterQuery<T>;
```

Comment:
Not sure why I am required to do this `casts`.  In Typescript 4.x, they are not needed, in 3.x it does not accept this is an argument for the below function.  I am open to ideas as to what may be wrong.

Reply:
I am not sure how to solve but what i think is wrong is:
this method save uses `data: WithPermissionsDataType<T>` while appendPermissionsQuery uses as the first param `query: PermissionsUwaziFilterQuery<T>`
the first is based on `PartialDataType`, which means, _id can be not defined, the second is based on `DataType` which means _id can NOT be undefined. i think despite the difficult to read error this is the issue.